File: src/gtm_sourcing_agent/stages/calibration.py
"""Stage 2: Hiring Manager Calibration. See prompts/calibration.md."""

import json

from .. import llm_client, storage
from ..models import HiringManagerCalibration


def run(role_id: str, *, storage_backend=storage) -> HiringManagerCalibration:
    jd = storage_backend.require_section(role_id, "job_description")
    prompt = llm_client.render_prompt("calibration.md", job_description_json=json.dumps(jd))
    result = llm_client.generate(prompt, HiringManagerCalibration, stage="calibration")
    storage_backend.merge_section(role_id, "calibration", result.model_dump())
    return result

File: backend-node/src/stages/calibration.ts
// Port of stages/calibration.py -- Stage 2: Hiring Manager Calibration.
import * as storage from "../db/storage.js";
import * as llmClient from "../llmClient.js";
import { HiringManagerCalibration } from "../models.js";

export async function run(roleId: string): Promise<HiringManagerCalibration> {
  const jd = await storage.requireSection(roleId, "job_description");
  const prompt = llmClient.renderPrompt("calibration.md", { job_description_json: JSON.stringify(jd) });
  const result = await llmClient.generate(prompt, HiringManagerCalibration, { stage: "calibration" });
  await storage.mergeSection(roleId, "calibration", result);
  return result;
}
